src/net/tcp.ts

```diff
@@ -116,6 +111,18 @@ export class Server {
       // TODO: investigate which other values occur
       this.fulfillShutdown_(SocketCloseKind.UNKOWN);
     }
+
+    // Since there may be other calls to this provider in progress which will
+    // fail to resolve if we destroy the communication channel now, postpone
+    // doing so until the next iteration of the event loop.
+    setTimeout(() => {
```

Comment:
You're right. Probably happenstance that this works right now. I couldn't think of a simple general solution...let me think about it some more.